crates/bevy_math/src/rotation2d.rs

```diff
@@ -0,0 +1,380 @@
+        Self {
+            cos: self.cos,
+            sin: -self.sin,
+        }
+    }
+
+    /// Performs a linear interpolation between `self` and `rhs` based on
+    /// the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s`
+    /// is `1.0`, the result will be equal to `rhs`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Rotation2d;
+    /// #
+    /// let rot1 = Rotation2d::IDENTITY;
+    /// let rot2 = Rotation2d::radians(std::f32::consts::FRAC_PI_2);
+    ///
+    /// let result = rot1.lerp(rot2, 0.5);
+    ///
+    /// assert_eq!(result.as_radians(), std::f32::consts::FRAC_PI_4);
+    /// ```
+    #[inline]
+    pub fn lerp(self, end: Self, s: f32) -> Self {
+        Self::from_sin_cos(self.sin.lerp(end.sin, s), self.cos.lerp(end.cos, s))
+    }
+
+    /// Performs a spherical linear interpolation between `self` and `end`
+    /// based on the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s`
+    /// is `1.0`, the result will be equal to `end`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Rotation2d;
+    /// #
+    /// let rot1 = Rotation2d::radians(std::f32::consts::FRAC_PI_4);
+    /// let rot2 = Rotation2d::radians(-std::f32::consts::PI);
+    ///
+    /// let result = rot1.slerp(rot2, 1.0 / 3.0);
+    ///
+    /// assert_eq!(result.as_radians(), std::f32::consts::FRAC_PI_2);
+    /// ```
+    #[inline]
+    pub fn slerp(self, end: Self, s: f32) -> Self {
+        self * Self::radians(self.angle_between(end) * s)
+    }
+}
+
+impl From<f32> for Rotation2d {
+    /// Creates a [`Rotation2d`] from a counterclockwise angle in radians.
+    fn from(rotation: f32) -> Self {
+        Self::radians(rotation)
+    }
+}
+
+impl From<Rotation2d> for Mat2 {
+    /// Creates a [`Mat2`] rotation matrix from a [`Rotation2d`].
+    fn from(rot: Rotation2d) -> Self {
+        Mat2::from_cols_array(&[rot.cos, -rot.sin, rot.sin, rot.cos])
+    }
+}
+
+impl std::ops::Mul for Rotation2d {
+    type Output = Self;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        Self {
+            cos: self.cos * rhs.cos - self.sin * rhs.sin,
+            sin: self.sin * rhs.cos + self.cos * rhs.sin,
+        }
+    }
+}
+
+impl std::ops::MulAssign for Rotation2d {
+    fn mul_assign(&mut self, rhs: Self) {
+        *self = *self * rhs;
+    }
+}
+
+impl std::ops::Mul<Vec2> for Rotation2d {
+    type Output = Vec2;
+
+    /// Rotates a [`Vec2`] by a [`Rotation2d`].
+    fn mul(self, rhs: Vec2) -> Self::Output {
+        Vec2::new(
+            rhs.x * self.cos - rhs.y * self.sin,
+            rhs.x * self.sin + rhs.y * self.cos,
+        )
+    }
+}
+
+impl std::ops::Mul<Direction2d> for Rotation2d {
+    type Output = Direction2d;
+
+    /// Rotates a [`Direction2d`] by a [`Rotation2d`].
+    fn mul(self, rhs: Direction2d) -> Self::Output {
+        Direction2d::new_unchecked(self * *rhs)
```

Comment:
For what value of normalized?

Reply:
I'm guessing it could cause some drift, especially if the rotation gets denormalized (due to e.g. successive operations when running physics), but this is also an issue for Quat and Dir3. It's kinda unavoidable unless we normalize after every rotation, which could be expensive